stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/RetrofitCall.kt

```diff
@@ -40,83 +46,78 @@ internal class RetrofitCall<T : Any>(
         call.cancel()
     }
 
-    override fun execute(): Result<T> {
-        return execute(call)
-    }
+    override fun execute(): Result<T> = runBlocking { await() }
 
     override fun enqueue(callback: Call.Callback<T>) {
-        enqueue(call) {
-            if (!canceled.get()) {
-                callback.onResult(it)
+        scope.launch {
+            enqueue(call) { result ->
+                scope.launch { notifyResult(result) { callback.onResult(it) } }
```

Comment:
What do you think about removing the duplicated `scope.launch` by changing the `callback` as a suspend function like the below?

```kotlin
private suspend fun enqueue(call: retrofit2.Call<T>, callback: suspend (Result<T>) -> Unit)
private suspend fun notifyResult(result: Result<T>, callback: suspend (Result<T>) -> Unit)
private suspend fun notifyResult(result: Result<T>, callback: suspend (Result<T>) -> Unit) =
        withContext(DispatcherProvider.Main) {
            result.takeUnless { canceled.get() }?.let { callback(it) }
        }

```

Reply:
Already updated